docs/deploy-rbd.md

```diff
@@ -32,51 +32,45 @@ Option | Default value | Description
 
 **Available environmental variables:**
 
 `HOST_ROOTFS`: rbdplugin searches `/proc` directory under the directory set by `HOST_ROOTFS`.
 
-`KUBERNETES_CONFIG_PATH`: if you use `k8s_configmap` as metadata store, specify
-the path of your k8s config file (if not specified, the plugin will assume
-you're running it inside a k8s cluster and find the config itself).
-
-`POD_NAMESPACE`: if you use `k8s_configmap` as metadata store,
-`POD_NAMESPACE` is used to define in which namespace you want
-the configmaps to be stored
-
 **Available volume parameters:**
 
 Parameter | Required | Description
 --------- | -------- | -----------
-`monitors` | one of `monitors`, `clusterID` or `monValueFromSecret` must be set | Comma separated list of Ceph monitors (e.g. `192.168.100.1:6789,192.168.100.2:6789,192.168.100.3:6789`)
-`monValueFromSecret` | one of `monitors`, `clusterID` or and `monValueFromSecret` must be set | a string pointing the key in the credential secret, whose value is the mon. This is used for the case when the monitors' IP or hostnames are changed, the secret can be updated to pick up the new monitors.
-`clusterID` | one of `monitors`, `clusterID` or `monValueFromSecret` must be set | String representing a Ceph cluster, must be unique across all Ceph clusters in use for provisioning, cannot be greater than 36 bytes in length, and should remain immutable for the lifetime of the Ceph cluster in use
+`clusterID` | yes | String representing a Ceph cluster, must be unique across all Ceph clusters in use for provisioning, cannot be greater than 36 bytes in length, and should remain immutable for the lifetime of the Ceph cluster in use
 `pool` | yes | Ceph pool into which the RBD image shall be created
 `imageFormat` | no | RBD image format. Defaults to `2`. See [man pages](http://docs.ceph.com/docs/mimic/man/8/rbd/#cmdoption-rbd-image-format)
 `imageFeatures` | no | RBD image features. Available for `imageFormat=2`. CSI RBD currently supports only `layering` feature. See [man pages](http://docs.ceph.com/docs/mimic/man/8/rbd/#cmdoption-rbd-image-feature)
-`csi.storage.k8s.io/provisioner-secret-name`, `csi.storage.k8s.io/node-publish-secret-name` | for Kubernetes | name of the Kubernetes Secret object containing Ceph client credentials. Both parameters should have the same value
-`csi.storage.k8s.io/provisioner-secret-namespace`, `csi.storage.k8s.io/node-publish-secret-namespace` | for Kubernetes | namespaces of the above Secret objects
+`csi.storage.k8s.io/provisioner-secret-name`, `csi.storage.k8s.io/node-publish-secret-name` | (no) for Kubernetes | name of the Kubernetes Secret object containing Ceph client credentials. Both parameters should have the same value
+`csi.storage.k8s.io/provisioner-secret-namespace`, `csi.storage.k8s.io/node-publish-secret-namespace` | (no) for Kubernetes | namespaces of the above Secret objects
```

Comment:
Would the admin ID/key ever be stored in a user's namespace? The example shows both the admin and user credentials in the same yaml whereas the in-tree driver only required the user credentials in the user's namespace.

Reply:
... I am also confused again since looking at ```K8sConfig```, it is hard-coded to pull secrets named ```ceph-cluster-<cluster-id>```. The documentation for ```external-provisioner``` [1] shows that the secrets specified here will be passed in ```CreateVolumeRequest.secrets```/```DeleteVolumeRequest.secrets```/```NodePublishVolumeRequest.secrets``` -- so to me it's confusing that some places are pulling secrets from ```ceph-cluster-<cluster-id>``` and others are getting overrides from the ```external-provisioner```?

[1] https://kubernetes-csi.github.io/docs/print.html#createdelete-volume-secret

Reply:
Resolved as per [comments](https://github.com/ceph/ceph-csi/pull/312#issuecomment-482631799) raised on the PR around the solution.